cmd/pebble/cmd_run.go

```diff
@@ -36,34 +37,92 @@ var longRunHelp = `
+	"USR2": syscall.SIGUSR2,
+}
+
+// defaultShutdownSignals is the list of signals listened to by default for
+// initiating shut down procedures.
+var defaultShutdownSignals = []os.Signal{
+	syscall.SIGTERM,
+	syscall.SIGINT,
+	syscall.SIGQUIT,
+}
+
 type cmdRun struct {
 	clientMixin
 
-	CreateDirs bool   `long:"create-dirs"`
-	Hold       bool   `long:"hold"`
-	HTTP       string `long:"http"`
-	Verbose    bool   `short:"v" long:"verbose"`
+	CreateDirs      bool   `long:"create-dirs"`
+	Hold            bool   `long:"hold"`
+	HTTP            string `long:"http"`
+	ShutdownSignals string `long:"shutdown-signals"`
+	Verbose         bool   `short:"v" long:"verbose"`
 }
 
 func init() {
 	addCommand("run", shortRunHelp, longRunHelp, func() flags.Commander { return &cmdRun{} },
 		map[string]string{
-			"create-dirs": "Create pebble directory on startup if it doesn't exist",
-			"hold":        "Do not start default services automatically",
-			"http":        `Start HTTP API listening on this address (e.g., ":4000")`,
-			"verbose":     "Log all output from services to stdout",
+			"create-dirs":      "Create pebble directory on startup if it doesn't exist",
+			"hold":             "Do not start default services automatically",
+			"http":             `Start HTTP API listening on this address (e.g., ":4000")`,
+			"shutdown-signals": "Symbolic signal(s) pebble listens to for shutting down. Defaults to TERM,INT & QUIT ",
+			"verbose":          "Log all output from services to stdout",
 		}, nil)
 }
 
+func processShutdownSignals(rawUserSignals string) ([]os.Signal, error) {
+	sigs := strings.Split(rawUserSignals, ",")
+	signals := make([]os.Signal, len(sigs))
+	i := 0
+	for _, sig := range sigs {
+		sysSig, exists := symbolicShutdownSignals[sig]
+		if !exists {
+			return signals, fmt.Errorf("unrecognised symbolic signal %q, valid  values: %v",
+				sig, validSymbolicSignals())
+		}
+		signals[i] = sysSig
+	}
+	return signals, nil
+}
+
+func validSymbolicSignals() []string {
+	keys := make([]string, len(symbolicShutdownSignals))
+
+	i := 0
+	for k := range symbolicShutdownSignals {
+		keys[i] = k
+		i++
+	}
+	return keys
+}
+
 func (rcmd *cmdRun) Execute(args []string) error {
 	if len(args) > 0 {
 		return ErrExtraArgs
 	}
 
+	var err error
+	shutdownSignals := defaultShutdownSignals
+	if rcmd.ShutdownSignals != "" {
+		shutdownSignals, err = processShutdownSignals(rcmd.ShutdownSignals)
+		if err != nil {
+			return err
+		}
+	}
+
 	sigs := make(chan os.Signal, 2)
-	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	// We always want to watch for defaultShutdownSignals otherwise default go
+	// handlers will watch them for us. By appending defaultShutdownSignals we
+	// run the risk of doubling up the signals from the user and our default
+	// collection. This doesn't matter to notify and the list is small so we
+	// don't worry about it.
+	signal.Notify(sigs, append(shutdownSignals, defaultShutdownSignals...)...)
```

Comment:
Instead of this, can we make use of `signal.Ignore`, either `signal.Ignore()` to ignore all, then call `signal.Notify(...)`, or just `signal.Ignore(SIGINT, SIGQUIT, SIGTERM)` and then Notify? I haven't tested this, but that seems cleaner than doubling up and having to test below in the handler.